runbot/runbot.py

```diff
@@ -1263,7 +1263,7 @@ def branch_info(branch):
             })
 
         # consider host gone if no build in last 100
-        build_threshold = max(build_ids) - 100
+        build_threshold = max(build_ids or [0]) - 100
```

Comment:
I am not talking about `None`. The problem is that the variable `build_ids` is initialized only in the `if repo:`-branch. Thus, if there is no repository in the database, the variable is undefined in line 1266. In addition to your change in line 1266, the variable should be initialized to an empty list before line 1198.